src/Propel/Generator/Model/Database.php

```diff
@@ -572,10 +589,12 @@ public function addSequence($sequence)
      */
     public function removeSequence($sequence)
     {
-        if ($this->sequences) {
-            if (false !== ($idx = array_search($sequence, $this->sequences))) {
-                unset($this->sequence[$idx]);
-            }
+        if (!$this->sequences) {
+
+            return;
```

Comment:
It made more sense to me if the "happy case" wasn't wrapped in something "conditional".